students/duanez2021/lesson03/mailroom_rev01.py

```diff
@@ -0,0 +1,218 @@
+            amt = donor_db[i][1][k]
+            print('{:<25}' '${:>15,.2f}'. format(name, amt))
+    #print ('\n')
+
+
+def create_required_report(db):
+    print('{0: <20}{1} {2: >12} {3} {4: ^10} {5} {6: >13}'.format('Donor Name','|', 'Total Given','|', 'Num Gifts','|','Average Gift'))
+    print ('{:''^66}'.format(' '))
+    t = ()
+    for i, j in enumerate(db):
+        name = donor_db[i][0]
+        t = (name, get_sum_donations(db, name), get_number_of_donations(db, name),get_avg_donation_amt(db, name))
+        strFormat = '{0: <20.20}${1: >13,.2f}  {2:^11}  ${3:> 13,.2f}'
+        print(strFormat.format(t[0], t[1], t[2], t[3]))
+    print('\n')
+# create_required_report(donor_db)
+
+
+def _create_required_report(db):
+    #print ('Donor Name          | Total Given | Num Gifts | Average Gift')
+    print('{0:_<20}{1}{2:_>12}{3}{4:_^10}{5}{6:_>13}'.format('Donor Name','|', 'Total Given','|', 'Num Gifts','|','Average Gift'))
+    print ('{:''^66}'.format(' '))
+    t = ()
+    for i, j in enumerate(db):
+        name = donor_db[i][0]
+        t = (name, get_sum_donations(db, name), get_number_of_donations(db, name),get_avg_donation_amt(db, name))
+        sp=25-len(name)
+        #print (name, len(name), sp)
+        #strFormat = '{:25}${:13,.2f}'    ' {:^3}     ${:>13,.2f}'
+        #strFormat = '{0:<}{1:<' + str(sp) + '}'
+        strFormat = '{0:_<20}${1:_>13,.2f}{2:_^10}${3:_>13,.2f}'
+        #print (strFormat)
+        #'{:25}{:5}${:13,.2f}'    ' {:^3}     ${:>13,.2f}'
+        #print(strFormat.format(t[0], t[1], t[2], t[3]))
+        print(strFormat.format(t[0], t[1], t[2], t[3]))
+    print('\n')
+ #_create_required_report(donor_db)
+
+
+# create_required_report(donor_db)
+# print('{:<25}'  ' {:<10}${:>15,.2f}'.format(t[i - 2], t[i - 1], t[i]))
+# get_sum_donations(donor_db, 'Paul Allen')
+# get_number_of_donations(donor_db, 'Paul Allen')
+# get_avg_donation_amt(donor_db, 'Paul Allen')
+# print ('\n')
+
+def get_sum_donations(db, donor_name):
+    sum_amt= 0.0
+    for i, j in enumerate(db):
+        if donor_name == db[i][0]:
+            for k, m in enumerate (db[i][1][:]):
+                sum_amt += db[i][1][k]
+    return sum_amt
+
+
+def get_number_of_donations(db, donor_name):
+    n = 0
+    for i, j in enumerate(db):
+        if donor_name == db[i][0]:
+            for k, m in enumerate (db[i][1][:]):
+                n += 1
+    return n
+
+
+def get_avg_donation_amt(db, donor_name):
+    avrg = 0
+    sm = get_sum_donations(db, donor_name)
+    n = get_number_of_donations(db, donor_name)
+    return round(sm/n, 2)
+
+# get_sum_donations(donor_db, 'Paul Allen')
+# get_number_of_donations(donor_db,  'Paul Allen')
+# get_avg_donation_amt(donor_db,  'Paul Allen')
+#  create_simple_report(donor_db)
+
+
+def list_donors(db):
+    for i, j in enumerate(donor_db):
+        print('{:<25}'.format( donor_db[i][0]))
+
+# list_donors(donor_db)
+
+
+def check_donor_name(db, s):
+    b = False
+    for i, j in enumerate(db):
```

Comment:
you don't need to use enumerate unless there is specific need to have the index, and in this use case there isn't


`for donor_name, donations in donor_db` is the appropriate way to loop over a list.